src/runtime/Symbol.h

```diff
@@ -21,31 +21,58 @@
 #define __EscargotSymbol__
 
 #include "runtime/PointerValue.h"
+#include "util/Vector.h"
+#include "util/TightVector.h"
 
 namespace Escargot {
 
 class VMInstance;
 
+typedef void (*SymbolFinalizer)(Symbol* self, void* data);
+
+struct SymbolFinalizerData : public gc {
+    SymbolFinalizerData()
+        : m_removedFinalizerCount(0)
+    {
+    }
+
+    size_t m_removedFinalizerCount;
+    TightVector<std::pair<SymbolFinalizer, void*>, GCUtil::gc_malloc_atomic_allocator<std::pair<SymbolFinalizer, void*>>> m_finalizer;
+};
+
 class Symbol : public PointerValue {
 public:
     explicit Symbol(Optional<String*> desc = nullptr)
         : m_typeTag(POINTER_VALUE_SYMBOL_TAG_IN_DATA)
         , m_description(desc)
+        , m_finalizerData(nullptr)
     {
+        // nullptr of desc represents `undefined`
     }
 
-    Optional<String*> description() const
+    String* descriptionString() const;
+    Value descriptionValue() const;
+
+    SymbolFinalizerData* finalizerData() const
     {
-        return m_description;
+        ASSERT(!!m_finalizerData);
```

Comment:
I don't see a `hasFinalizerData`, is there a way to check this somehow?

Reply:
This `removeFinalizer` will be called when it has a valid data by GC (it is guaranteed)